cargo-shuttle/src/lib.rs

```diff
@@ -670,6 +666,34 @@ impl Shuttle {
         Ok(())
     }
 
+    async fn add_runtime_info(
+        runtime: Option<(
+            Child,
+            RuntimeClient<ClaimService<InjectPropagation<Channel>>>,
+        )>,
+        existing_runtimes: &mut Vec<(
+            Child,
+            RuntimeClient<ClaimService<InjectPropagation<Channel>>>,
+        )>,
+        provisioner_server: &JoinHandle<Result<(), tonic::transport::Error>>,
+    ) -> Result<(), Status> {
+        match runtime {
+            Some(inner) => existing_runtimes.push(inner),
+            None => {
+                // Stopping all runtimes gracefully and then forcefully because we must follow up with a `std::process:exit`
+                // which doesn't guarantee destructors run, which means dropped runtimes don't result in killed runtimes (per
+                // https://docs.rs/tokio/latest/tokio/process/struct.Child.html#caveats).
+                for rt_info in existing_runtimes {
```

Comment:
I had to do a mutable borrow here because:
1) We can not do destructuring because we can not destructure and mutable borrow at the same time the values within the tuple, e.g. this breaks compilation:
```rust
for (&mut rt, &mut rt_client) in existing_runtimes {
...
}
```

while simply moving out the tuple contents from a mutable reference (existing runtimes is a `&Vec<(Child, RuntimeClient)>`) breaks the compilation again:
```
for (mut rt, mut rt_client) in existing_runtimes {
...
}
```